src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,194 @@
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(get(resultSet.getObject(1, Long.class)).orElseThrow(() ->
+                        new DataProcessingException("Can't find car for driver with such id: "
+                                + driverId)));
+            }
+            return cars;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get all cars for driver with id: "
+                    + driverId, throwable);
+        }
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        manufacturer.setId(manufacturerId);
+        Long carId = resultSet.getObject(1, Long.class);
+        String carModel = resultSet.getString("model");
+        Car car = new Car(carModel, manufacturer);
+        car.setId(carId);
+        return car;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String query = "SELECT id, name, license_number FROM drivers d "
+                + "JOIN cars_drivers c_d ON d.id = c_d.driver_id "
+                + "WHERE car_id = ? AND d.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getByIdStatement = connection.prepareStatement(query)) {
+            getByIdStatement.setLong(1, carId);
+            ResultSet resultSet = getByIdStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(getDriverFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get car from DB with id: " + carId, throwable);
+        }
+    }
+
+    private Driver getDriverFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String name = resultSet.getString("name");
+        String licenseNumber = resultSet.getString("license_number");
+        return new Driver(id, name, licenseNumber);
+    }
+
+    private void saveDrivers(Car car) throws SQLException {
+        String query = "INSERT cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement = connection.prepareStatement(query)) {
+            insertStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertStatement.setLong(2, driver.getId());
+                insertStatement.executeUpdate();
+            }
```

Comment:
It is better to use catch here then you will see the exception with descriptive message that error happened when saving drivers but not trying to save the car.